File: solutions/csharp/allergies/Allergies.cs
public enum Allergen
{
    Eggs         = 1,
    Peanuts      = 2,
    Shellfish    = 4,
    Strawberries = 8,
    Tomatoes    = 16,
    Chocolate   = 32,
    Pollen      = 64,
    Cats        = 128
}

public class Allergies
{
    public Allergies(int mask)
    {
        m_mask = mask;
    }

    public bool IsAllergicTo(Allergen allergen)
    {
        return (m_mask & ((int)allergen)) != 0;
    }

    public Allergen[] List()
    {
        var result = new List<Allergen>();

        foreach (Allergen a in Enum.GetValues<Allergen>())
        {
            if (IsAllergicTo(a))
                result.Add(a);
        }

        return [.. result];
    }

    private readonly int m_mask;
}

File: solutions/julia/allergies/allergies.jl
const allergen_scores = Dict(
    "eggs"         =>   1,
    "peanuts"      =>   2,
    "shellfish"    =>   4,
    "strawberries" =>   8,
    "tomatoes"     =>  16,
    "chocolate"    =>  32,
    "pollen"       =>  64,
    "cats"         => 128
)

function allergic_to(score, allergen)
    return score & allergen_scores[allergen] != 0
end

function allergy_list(score)
    return Set([p[1] for p in allergen_scores if score & allergen_scores[p[1]] != 0])
end
